### ISSUE TITLE
clone3() returns EINVALID if exit_signal set

### ISSUE THREAD

**Issue by adrianreber** [checkpoint-restore/criu#926]
Trying to run CRIU to restore a runc based container does not work as clone3() fails.

Looking at the kernel there is a check if CLONE_PARENT is set and exit_signal is set and if both are set the kernel returns EINVALID.

Talking with @brauner about this he mentioned that the kernel always ignored the exit_signal if CLONE_PARENT is set even with clone(). The exit_signal of the thread group leader will always be used if CLONE_PARENT is specified.

CLONE_PARENT is mentioned a few times and I am not sure how to correctly handle it. My quick fix is to just unset `exit_signal` if CLONE_PARENT is set. But only for clone3():
```diff
-       c_args.exit_signal = exit_signal;
+       if (!(flags && CLONE_PARENT))
+               c_args.exit_signal = exit_signal;
```

As CLONE_PARENT seems to be necessary, according to the comments in `cr-restore.c` to work around some bugs in older kernels, this seems to be the best fix right now.

@avagin in other ideas how solve this best?

**Comment by brauner** [comment]
Do you pass anything other than SIGCHLD somewhere?

**Comment by adrianreber** [comment]
A quick grep through the code seems to indicate it is only SIGCHLD.

**Comment by adrianreber** [comment]
Possible fix that works for me: #930

**Comment by adrianreber** [comment]
Fixed with #930
